### PR TITLE
SageMaker TrainingJob Controller with e2e Tests in Python

### PR DESCRIPTION
**Pull request by mbaijal**
## Description of changes:
Controller for SageMaker TrainingJob resource with integration test. 
 - Updated generator.yaml config file with a specific operation for `StopTrainingJob` and various Status fields
 - Added the required message_prefix to the `404 error Code` for trainingJob.
 - Added 4 Status Fields for the trainingJob. Verified that the non-zero fields gets populated on `kubectl describe` and are in sync with the SageMaker Status. 
 - code-generated CRD, controller, rbac and manifests to help create installers.
 - e2e tests to create and stop a trainingjob using the new Python test framework.
 - Relevant dataset has been added to the bucket.
 
## Pending:
 - Add a trainingJob Test for debugger
 - [Maybe] Remove the sleep from the test_trainingjob.py script and use a wait loop instead.
 
## Testing
Using personal account, I have tested this using the manual step by step invocation as described in the Python Test README. Fully automated invocation did not work for me. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jaypipes — APPROVED** [review]
rock on, @mbaijal this is awesome.

File: services/sagemaker/generator.yaml [hunk 1]
```diff
@@ -1,10 +1,41 @@
...
           message_prefix: Could not find model
+  TrainingJob:
+    exceptions:
+      errors:
+          404:
+            code: ValidationException
+            message_prefix: Requested resource not found
+    fields:
+      TrainingJobStatus:
+          is_read_only: true
```
Comment (RedbackThomson) [inline services/sagemaker/generator.yaml:20]: Add `is_printable` for `TrainingJobStatus`, `SecondaryStatus` and `FailureReason`

Comment (mbaijal) [reply services/sagemaker/generator.yaml:20]: oh yea, I noticed that was missing. Let me create a separate Pr for that.

Comment (surajkota) [reply services/sagemaker/generator.yaml:20]: what do these mean, is_read_only, is_printable (probably printer columns)

**Pull request merged**